### PR TITLE
1) Support for new frameworks and bug fixes

### PR DESCRIPTION
**Pull request by anurag1408**
1) Support for Appium python
2) bug fix for pulling the device farm icon
3) Fix passing filter tags for UiAutomator
4) Fix passing filer tags for Instrumentation
5) Add support for passing calabash profile

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmRecorder.java [hunk 1]
```diff
@@ -86,6 +94,9 @@
     // XCTest
     public String xctestArtifact;
 
+    // XCTest
```
Comment (ahawker) [inline src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmRecorder.java]: *XCTest UI (XCUI)

File: src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmRecorder.java [hunk 2]
```diff
@@ -708,6 +771,15 @@ private boolean validateTestConfiguration() {
 
                 break;
             }
+            
+            case XCTEST_UI: {
+                if (xctestUiArtifact == null || xctestUiArtifact.isEmpty()) {
+                    writeToLog("XC UI tests artifact must be set.");
```
Comment (ahawker) [inline src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmRecorder.java]: *XC UI -> XCTest UI

File: src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmUploadType.java [hunk 3]
```diff
@@ -11,11 +11,13 @@
     IOS_APP ("IOS_APP"),                                    // App
     APPIUM_JAVA_TESTNG("APPIUM_JAVA_TESTNG_TEST_PACKAGE"),  // Test
     APPIUM_JAVA_JUNIT("APPIUM_JAVA_JUNIT_TEST_PACKAGE"),    // Test
+    APPIUM_PYTHON("APPIUM_PYTHON_TEST_PACKAGE"),  // Test
```
Comment (ahawker) [inline src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmUploadType.java]: Fix comment alignment.

**Comment by ahawker** [comment]
I appears you're missing a "doCheckXCTestUI" style method for form validation.

**Comment by ahawker** [comment]
Does your log changes address the same problem addressed in #30 ?

**Comment by anurag1408** [comment]
>> I appears you're missing a "doCheckXCTestUI" style method for form validation. 
Added UI validation for both XCTest and XCTest UI

>> Does your log changes address the same problem addressed in #30 ?
Yes

Also incorporated your other review comments

**Pull request merged**